Ragam search

shaaza opened the pull request:
Displays ragam search results from the API in a table.

shaaza committed:
Add README and project.clj

shaaza committed:
Add .gitignore and html and css resources.

shaaza committed:
Ragam search bar
    - core: Contains root view, plumbing to render the root view, and dev configuration. Routes are hooked into root view.
    - routes: Contains the routes for client-side routing.
    - views: Contains all the reagent components.
    - subs: Contains the initial app-db and subscriptions.
    - events: Contains event handlers, effectful and otherwise.
    - effects: Contains effects, and also a var to declare the API URL to make calls to.

shaaza committed:
Show search results and loading text.

shaaza committed:
Use Reactable Table to display search results.

shaaza committed:
Basic CSS styling using framework.

ssrihari commented on src/cljs/app/effects.cljs:
```diff
@@ -0,0 +1,17 @@
+              :handler         #(re-frame/dispatch [:search/receive-results %])
+              :response-format :json
+              :keywords?       true
+              })))
```
`http-client` might be a better name for this ns at the moment. Even if it is a side-effect, naming it from a domain perspective makes it more recognizable.

ssrihari commented on src/cljs/app/events.cljs:
```diff
@@ -0,0 +1,34 @@
+   {:db (assoc db :loading true)
+    :http-get {:endpoint "search"
+               :query query
+               :type "ragam"}}))
```
Would be good to pull out these functions, and give them names. Makes them more accessible, usable in the repl, and testable.

You can write these like so:
```clojure
(defn search-ragams [{:keys [db]} [_ query]]
   {:db (assoc db :loading true)
    :http-get {:endpoint "search"
               :query query
               :type "ragam"}})

(defn init []
 (re-frame/reg-event-fx :search/ragams! search-ragams)
 (re-frame/reg-event-fx :another-event another-event))
 ```

I'd suggest trying to keep all the code within functions wherever possible (like in `init` above). This gives better control over when they're run. "When this ns is required" is a rather crude, potentially confusing way of running code.

ssrihari commented on src/cljs/app/subs.cljs:
```diff
@@ -0,0 +1,29 @@
+(re-frame/reg-sub
+ :search/results
+ (fn [{db :search}]
+   (:results db)))
```
Similar to https://github.com/carnatic-music/kosha-ui/pull/1/files#r109340411.

shaaza committed:
Converted view to Type 1 components and put components in a vector.

shaaza committed:
Renamed effects ns and pulled out handlers & subs as functions.
- Renamed kosha.app.effects to kosha.app.http-client
- Pulled out event handlers and subscriptions as functions.
- Separated registration of handlers/subs and invoked them explicitly on startup.

shaaza committed:
Move Reactable require to the (only) ns where it's used: views.
